src/utils/base.jl

```diff
@@ -16,6 +16,7 @@ export nframes,
     unflatten_batch
 
 using StatsBase
+using ElasticArrays: ElasticArray
 
```

Comment:
I think `ElasticArray` related changes are not necessary here. Could you describe why you need them?

Reply:
You are right.
I made a mistake.

```
julia> discount_rewards(ElasticArray([1f0, 2, 3]), 0.9; dims = 1)
ERROR: MethodError: no method matching _discount_rewards!(::ElasticArray{Float64,1,0,Array{Float64,1}}, ::ElasticArray{Float32,1,0,Array{Float32,1}}, ::Float64, ::Nothing, ::Nothing, ::Int64)
Closest candidates are:
  _discount_rewards!(::AbstractArray{T,1} where T, ::AbstractArray{T,1} where T, ::Any, ::Any, ::Any, ::Colon) at D:\x\.julia\dev\ReinforcementLearningCore\src\utils\base.jl:231
  _discount_rewards!(::Any, ::Any, ::Any, ::Any, ::Nothing) at D:\x\.julia\dev\ReinforcementLearningCore\src\utils\base.jl:234
  _discount_rewards!(::Any, ::Any, ::Any, ::Any, ::Any) at D:\x\.julia\dev\ReinforcementLearningCore\src\utils\base.jl:237
  ...

```

This is the vector type, and does not have "dims" parameter.

these use cases are OK
```
julia> discount_rewards(ElasticArray([1f0 2 3; 4 5 6]), 0.9; dims=2)
2×3 ElasticArray{Float64,2,1,Array{Float64,1}}:
  5.23   4.7  3.0
 13.36  10.4  6.0

julia> discount_rewards(ElasticArray([1f0, 2, 3]), 0.9)
3-element ElasticArray{Float64,1,0,Array{Float64,1}}:
 5.23
 4.7
 3.0
```